File: Sharebook/Client/src/app/core/Core/Models/TwitterEntities/HashTagEntity.ts
import {IHashtagEntity} from "../../../Public/Models/Entities/IHashTagEntity";

// A hashtag is a keyword prefixed by # and representing a category of tweet
// This class stores information related with an user mention
export class HashtagEntity implements IHashtagEntity {
  // [JsonProperty("text")]
  public text: string;

  // [JsonProperty("indices")]
  public indices: number[];

  public equals(other: IHashtagEntity): boolean {
    if (other == null || this.text !== other.text) {
      return false;
    }

    if (this.indices == null || other.indices == null) {
      return this.indices === other.indices;
    }

    // @ts-ignore
    return this.indices.containsSameObjectsAs<number>(other.indices, true);
  }

  public toString(): string {
    return `#${this.text}`;
  }
}

File: Server/src/Chessbook.Core/Domain/Post/Entities/HashTagEntity.cs
namespace Chessbook.Data.Models.Post.Entities
{
    using Chessbook.Data.Common.Models;

    /// <summary>
    /// A hashtag is a keyword prefixed by # and representing a category of tweet
    /// This class stores information related with an user mention
    /// </summary>
    public class HashtagEntity : BaseDeletableModel<long>
    {
        public string Text { get; set; }

        public int IndicesId { get; set; }
        public Indices Indices { get; set; }

        //public bool Equals(IHashtagEntity other)
        //{
        //    if (other == null || Text != other.Text)
        //    {
        //        return false;
        //    }

        //    if (Indices == null || other.Indices == null)
        //    {
        //        return Indices == other.Indices;
        //    }

        //    return Indices.ContainsSameObjectsAs(other.Indices, true);
        //}

        //public override string ToString()
        //{
        //    return $"#{Text}";
        //}
    }
}
